src/memory.cpp

```diff
@@ -54,6 +66,15 @@ MemBlock SingleArrayMemory::getMemBlock(const expr &bid) const {
   return MemBlock(array, writable, numelem);
 }
 
+expr SingleArrayMemory::addLocalBlock(const expr &numelem, const expr &writable) {
+  assert(numLocalBlocks < maxLocalBlocks);
```

Comment:
New local block can be created only when numLocalBlocks is sticky lower than maxLocalBlocks.